pkg/services/ngalert/schedule.go

```diff
@@ -50,11 +50,17 @@ func (ng *AlertNG) definitionRoutine(grafanaCtx context.Context, key alertDefini
 				results, err := eval.ConditionEval(&condition, ctx.now)
 				end = timeNow()
 				if err != nil {
+					// consider saving alert instance on error
 					ng.schedule.log.Error("failed to evaluate alert definition", "key", key, "attempt", attempt, "now", ctx.now, "duration", end.Sub(start), "error", err)
 					return err
 				}
 				for _, r := range results {
 					ng.schedule.log.Info("alert definition result", "key", key, "attempt", attempt, "now", ctx.now, "duration", end.Sub(start), "instance", r.Instance, "state", r.State.String())
```

Comment:
Now that we save these, I think this can change to Debug so as not to add to many line to the log.

Could collect summary for all results within a run, and log a single info line summary (X instances, N alerting, O Normal).

Reply:
Also might be nice to have definition name in log as well (as UIDs are hard to read/keep track of)